### ISSUE TITLE
Python v3.11 Update broke other programs?

### ISSUE THREAD

**Issue by prashant-hm** [void-linux/void-packages#40463]
I think the recent update of Python v3.11 has broken a few programs.
For instance, Deluge.

Below is the output that I get now. This was not the case a few weeks back. I understand there is an annual update for Python is happening (5-7 days back). The [waterfall](https://build.voidlinux.org/waterfall) seems to be showing the x86_64 build complete.

I raised this issue on Reddit, where one more user also faced breakage of some other issue. (https://www.reddit.com/r/voidlinux/comments/yrzpee/python_311_update_has_it_broken_some_programs/). I have another program "castero" which started giving error due to python.

![image](https://user-images.githubusercontent.com/563296/201291108-a1486bfa-f7a0-4357-9daa-7814d4d2325d.png)

Any help will be highly appreciated!

**Comment by sgn** [comment]
For deluge, can you try update boost to `1.80.0_6`?

**Comment by prashant-hm** [comment]
![image](https://user-images.githubusercontent.com/563296/201307651-7efac9f8-002d-4afb-b9f2-fdd746dad2b5.png)

I think it's already there.

**Comment by sgn** [comment]
`xbps-query -p pkgver boost-python3-1.80`? And try start deluge again?

**Comment by prashant-hm** [comment]
> `xbps-query -p pkgver boost-python3-1.80`? And try start deluge again?

![image](https://user-images.githubusercontent.com/563296/201311802-ecbc4475-4c50-41ac-9879-cb8049b3c904.png)

Starting deluge again, gives same issue still.

**Comment by sgn** [comment]
Hm, it works in my machine. Let's me bump libtorrent-rasterbar, then

**Comment by sgn** [comment]
Anyway, please don't post image, just copy text is enough.

**Comment by simplyTedel** [comment]
I get this error when trying to run rednotebook (version 2.19_4)

````
Adding /usr/lib/python3.11/site-packages to sys.path
Traceback (most recent call last):
  File "/usr/bin/rednotebook", line 6, in <module>
    import journal
ModuleNotFoundError: No module named 'journal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/rednotebook", line 11, in <module>
    import rednotebook.journal
  File "/usr/lib/python3.11/site-packages/rednotebook/journal.py", line 88, in <module>
    elibintl.install(GETTEXT_DOMAIN, LOCALE_PATH, libintl=None)
  File "/usr/lib/python3.11/site-packages/rednotebook/external/elibintl.py", line 472, in install
    _install(domain, localedir, True, libintl=libintl)
  File "/usr/lib/python3.11/site-packages/rednotebook/external/elibintl.py", line 431, in _install
    gettext.bind_textdomain_codeset(domain, 'UTF-8')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'gettext' has no attribute 'bind_textdomain_codeset'
```

**Comment by mtroberts** [comment]
I get the following error with Syncplay:

```
Traceback (most recent call last):
  File "/usr/lib/syncplay/syncplay/client.py", line 837, in start
    waitForConnection = self._reconnectingService.whenConnected(failAfterFailures=1)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/twisted/application/internet.py", line 1168, in whenConnected
    return self._machine.whenConnected(failAfterFailures)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_methodical.py", line 232, in __get__
    @preserveName(self.method)
     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 43, in decorator
    return copyfunction(decorated,
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 35, in copyfunction
    return function(copycode(template.__code__, codechanges), *values)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 23, in copycode
    return code(*values)
           ^^^^^^^^^^^^^
TypeError: code() argument 13 must be str, not int

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/bin/syncplay", line 19, in <module>
    ep_client.main()
  File "/usr/lib/syncplay/syncplay/ep_client.py", line 8, in main
    SyncplayClientManager().run()
  File "/usr/lib/syncplay/syncplay/clientManager.py", line 16, in run
    syncplayClient.start(config['host'], config['port'])
  File "/usr/lib/syncplay/syncplay/client.py", line 839, in start
    waitForConnection = self._reconnectingService.whenConnected()
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/twisted/application/internet.py", line 1168, in whenConnected
    return self._machine.whenConnected(failAfterFailures)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_methodical.py", line 232, in __get__
    @preserveName(self.method)
     ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 43, in decorator
    return copyfunction(decorated,
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 35, in copyfunction
    return function(copycode(template.__code__, codechanges), *values)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/automat/_introspection.py", line 23, in copycode
    return code(*values)
           ^^^^^^^^^^^^^
TypeError: code() argument 13 must be str, not int
```

**Comment by paper42** [comment]
@simplyTedel @mtroberts could you open separate issues for each of your issues? This issue already contains a lot of information and it will help us track the issues without losing them.

**Comment by prashant-hm** [comment]
> > > > [void-linux/void-packages#40453](https://github.com/void-linux/void-packages/pull/40453), which other programs are broken?
> > > 
> > > 
> > > [*] castero-0.9.4_3 TUI podcast client for the terminal
> > > And...one that the fellow Redditor posted....
> > > focuswriter
> > 
> > 
> > Can you post the backtraces printed in the terminal for both of them?
> 
> Here the one for castero:
> 
> ![image](https://user-images.githubusercontent.com/563296/201294417-fa3bc373-c933-42fe-8e1d-628407650772.png)
> 
> For "focuswriter" (the package that other Redditor had issue with), I am not facing any issue. Just installed and ran it. It's working fine.
> 
> Thanks!

I can confirm, castero package is working now. Thanks a lot!
Deluge is still down.

**Comment by fanyx** [comment]
`libpyldb-util` and `libpytalloc-util` are broken, too
Required for samba

**Comment by paper42** [comment]
> `libpyldb-util` and `libpytalloc-util` are broken, too
> Required for samba

Please open separate issues and describe what's broken and how to reproduce it.

**Comment by simplyTedel** [comment]
> @simplyTedel could you open separate issues for each of your issues? This issue already contains a lot of information and it will help us track the issues without losing them.

It might not be neccesary for RedNotebook's case. I could read on Python's release notes that "Function bind_textdomain_codeset() (…) are also removed, since they are only used for the l*gettext() functions." [Source](https://docs.python.org/uk/3.11/whatsnew/3.11.html).

I understand the issue is on their side, and that it was not caused by this update. As [a fellow Reddit user points out](https://www.reddit.com/r/voidlinux/comments/yshqvi/comment/ivzymn6/), we need to wait for an update.

**Comment by paper42** [comment]
@simplyTedel fixed in 7707e3eaa3660ca136511aa776405c6dd70fd82f, but it would still be nice to have one issue per problem if there is not enough of them for a tracking issue imho.

**Comment by sgn** [comment]
@prashant-hm for deluge, I can't reproduce. I thought it's about libtorrent-rasterbar need to be rebuild (depsite that I can run deluge without it rebuilt, and the author of https://github.com/void-linux/void-packages/pull/40453 confirmed no need to rebuild libtorrent-rasterbar). However upgrading doesn't solve your case. Can your provide more info about your system? Did you hold any packages?

**Comment by prashant-hm** [comment]
> @prashant-hm for deluge, I can't reproduce. I thought it's about libtorrent-rasterbar need to be rebuild (depsite that I can run deluge without it rebuilt, and the author of #40453 confirmed no need to rebuild libtorrent-rasterbar). However upgrading doesn't solve your case. Can your provide more info about your system? Did you hold any packages?

@sgn Thanks for taking time and effort to resolve the issue.
Does not look like the libtorrent-rasterbar rebuild had any effect. I have been updating my machine using "xbps-install -Syu". So, we may rule that out.
I removed the Deluge package from my machine with "xbps-remove". And then, "xbps-remove --remove-orphans".
Then installed it again using "xbps-install deluge deluge-gtk". Same issue.

The following packages are getting installed,
Name                         Action    Version           New version            Download size
boost-python3-1.80           install   -                 1.80.0_6               -
libtorrent-rasterbar         install   -                 1.2.18_1               -
libtorrent-rasterbar-python3 install   -                 1.2.18_1               -
python3-Mako                 install   -                 1.1.3_5                -
python3-automat              install   -                 22.10.0_1              -
python3-constantly           install   -                 15.1.0_7               -
python3-hyperlink            install   -                 21.0.0_3               -
python3-incremental          install   -                 21.3.0_2               -
python3-ply                  install   -                 3.11_7                 -
python3-pycparser            install   -                 2.21_1                 -
python3-cffi                 install   -                 1.15.1_2               -
python3-cryptography         install   -                 38.0.3_1               -
python3-openssl              install   -                 22.1.0_1               -
python3-pyasn1               install   -                 0.4.8_5                -
python3-pyasn1-modules       install   -                 0.2.8_6                -
python3-service_identity     install   -                 18.1.0_6               -
python3-typing_extensions    install   -                 4.2.0_2                -
python3-Twisted              install   -                 22.10.0_1              -
python3-chardet              install   -                 5.0.0_2                -
python3-rencode              install   -                 1.0.6_9                -
python3-setproctitle         install   -                 1.2.1_3                -
python3-xdg                  install   -                 0.28_2                 -
deluge                       install   -                 2.1.1_1                -
deluge-gtk                   install   -                 2.1.1_1                -

The version of python on my machine is: v3.11.0. This is not running in any virtual python env.

Not sure what you meant by "hold any packages". If you meant if I compiled any from source, the answer is No.

Thanks once again!

**Comment by sgn** [comment]
> Not sure what you meant by "hold any packages". If you meant if I compiled any from source, the answer is No.

I meant:
```sh
xbps-query -H
```

**Comment by prashant-hm** [comment]
> I meant:
> 
> ```shell
> xbps-query -H
> ```
This did not yield any output.

> 
> Maybe, you can show us this as well:
> 
> ```
>  xbps-query -l | awk '{print $2}' | xargs -L1 xbps-query -p repository| sort -u
> ```
/hostdir/binpkgs
https://alpha.de.repo.voidlinux.org/current
https://repo-default.voidlinux.org/current
https://repo-default.voidlinux.org/current/nonfree

"/hostdir/binpkgs" looks odd. I tried to see what is "/hostdir/binpkgs", by doing "ls /hostdir/binpkgs" and got,
ls: cannot access '/hostdir/binpkgs': No such file or directory

**Comment by sgn** [comment]
```sh
xbps-query -l | awk '{print $2}' | xargs -I'{}' sh -c "xbps-query -p repository,pkgver {} | paste -s | awk '\$1 == \"/hostdir/binpkgs\" { print \$2 }'"
```

Try to remove those packages and re-install Void Linux version.

**Comment by prashant-hm** [comment]
@sgn 

This seems to be specific to my local setup. 
I am guessing the python3.11 links to libraries (libtorrent) were all messed up on my machine. I am saying so because I tried the following,

`$ python -c "import libtorrent; print(libtorrent.version)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
SystemError: initialization of libtorrent raised unreported exception`

Anyways, I uninstalled the deluge package first, "xbps-remove deluge deluge-gtk".

Then I compiled Deluge from its Git repo source code. Build and installed it. This seems to have fixed all the library linkages. And Deluge started to work.

However, since I wanted the Void repo version of Deluge (just for the convenience of updates), I uninstalled Deluge.
For people unfamiliar with uninstall process, you can do `sudo updatedb; locate deluge` to figure the locations and remove it. Please refer to Deluge git page for uninstallation instructions.

Then I did usual install using "xbps-install deluge deluge-gtk". It has started working.

It will be interesting to see how things work out in the next package update.

Thanks! We can close this issue now.

**Comment by sgn** [comment]
Sorry, but what is the problem? Are you installing your own libtorrent/deluge and it's messed up? Is your system working now? with Void Linux's deluge? Did you do anything else? Since there's a mention of `/hostdir/binpkgs`, there must be some packages you're installing yourself and it's the problem.

Am I correct?

**Comment by prashant-hm** [comment]
> Sorry, but what is the problem? 
I could not pin-point the issue here.

> Are you installing your own libtorrent/deluge and it's messed up? 
Initially I had installed the deluge package from void repo. That was giving issue. I did not install it from source.

Installing Deluge from git source was just a work around. That worked.

> Is your system working now? 
Yes, deluge is working on Void Linux now.

>with Void Linux's deluge? 
Yes.

> Did you do anything else? 
No. All that I did has been mentioned above. Nothing else.

> Since there's a mention of `/hostdir/binpkgs`, there must be some packages you're installing yourself and it's the problem.
> Am I correct?
In last many months, I have not tried to manually/source install any packages myself. So, I don't think breakage is due to that. The existence of `/hostdir/binpkgs` may be due to a very old install. But deluge was used few weeks back too.

Thanks!
